### PR TITLE
submission page loop fix

### LINKED ISSUE #1066: Cannot reach submission page after failing to submit answers
### Expected behaviour
Given I have completed the census_household.json survey
And my answers have failed to submit because the queue is down
When I try to submit my answers again
Then the submission is successful

### Actual behaviour
Given I have completed the census_household.json survey
And my answers have failed to submit because the queue is down
When I try to submit my answers again
Then I cannot get to the confirmation page to resubmit

User will appear to be in a never ending loop

### Steps to reproduce the behaviour
1. Start survey runner locally with `EQ_RABBITMQ_ENABLED=True` or not set at all to simulate a failing message queue assuming you're not running RabbitMQ locally.
1. Load census_household.json on http://localhost:5000/dev
1. Complete the census questionnaire
1. Submit your answers expecting a 503
1. Stop/start the survey runner locally with `EQ_RABBITMQ_ENABLED=False` to simulate a working message queue
1. Attempt to complete the questionnaire

#### Browser
N/A

#### Operating System
N/A

### Screenshot
image

### PR DESCRIPTION
**Pull request by yoganandkunche**
### What is the context of this PR?
If the submission failed the user will be left in an infinite loop. 

fixes #1066 

### How to review 
Retest #1066

### CODE REVIEW

**Comment by benjefferies** [comment]
@yoganandkunche your build is failing

File: tests/app/questionnaire/test_path_finder.py [hunk 1]
```diff
@@ -965,3 +965,40 @@ def test_build_path_with_invalid_location(self):
         )
 
         self.assertEqual([], path_finder.build_path(blocks, invalid_block_location))
+
+    def test_given_no_completed_blocks_when_get_latest_location_then_get_to_first_block(self):
```
Comment (LJBabbage) [inline tests/app/questionnaire/test_path_finder.py]: 'go to first block' rather then 'get to first block'?

File: app/questionnaire/path_finder.py [hunk 2]
```diff
@@ -191,6 +191,8 @@ def get_latest_location(self, completed_blocks=None, routing_path=None):
 
             if incomplete_blocks:
                 return incomplete_blocks[0]
+            else:
+                return routing_path[-1]
 
         return routing_path[0]
```
Comment (LJBabbage) [inline app/questionnaire/path_finder.py]: Not a fan of the multiple returns

**Review by LJBabbage — CHANGES_REQUESTED** [review]

**Review by collisdigital — APPROVED** [review]
Tested locally before and after the fix by enabling RabbitMQ settings; works.

File: tests/app/questionnaire/test_path_finder.py (hunk 1 above)
Comment (yoganandkunche) [reply tests/app/questionnaire/test_path_finder.py]: Addressed. Please review

File: app/questionnaire/path_finder.py (hunk 2 above)
Comment (yoganandkunche) [reply app/questionnaire/path_finder.py]: Addressed. Please review

**Review by LJBabbage — APPROVED** [review]

**Comment by LJBabbage** [comment]
Squash and rebase please

File: tests/app/questionnaire/test_path_finder.py [hunk 3]
```diff
@@ -965,3 +965,40 @@ def test_build_path_with_invalid_location(self):
...
+        # Given no completed blocks
+        survey = load_schema_file("test_repeating_household.json")
+        path_finder = PathFinder(survey)
+        completed_block = []
+
+        # When get latest location
+        latest_location = path_finder.get_latest_location(completed_block)
+
+        # Then get to the first block
```
Comment (LJBabbage) [inline tests/app/questionnaire/test_path_finder.py]: Still have some typos in here

**Review by LJBabbage — CHANGES_REQUESTED** [review]

File: tests/app/questionnaire/test_path_finder.py (hunk 3 above)
Comment (yoganandkunche) [reply tests/app/questionnaire/test_path_finder.py]: made the changes, please review

**Review by LJBabbage — APPROVED** [review]

**Pull request merged**